File: src/algorithms/binarySearch.ts
export default function binarySearch(array: number[], target: number): number {
	let low = 0;
	let high = array.length - 1;

	while (low <= high) {
		const mid = Math.floor((low + high) / 2);

		if (array[mid] < target) low = mid + 1;
		else if (array[mid] > target) high = mid - 1;
		else return mid;
	}

	return -1;
}

File: tests/algorithms/search/binary_search.cpp
#include <array>
#include <print>

import binary_search;

auto main() -> int {
	auto array = std::array{1, 2, 3, 4, 5, 6};

	const auto* it = binary_search(array, 3);

	if (it != array.end()) std::println("found");
	else std::println("not found");
}
